src/pages/ThorChainLP/Pool/components/PoolChart.tsx

```diff
@@ -1,34 +1,130 @@
-import { RawText } from 'components/Text'
+import { thorchainAssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import type { SingleValueData } from 'lightweight-charts'
+import { useCallback, useMemo, useState } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Interval } from 'react-queries/queries/midgard'
+import { SimpleChart } from 'components/SimpleChart'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import type {
+  MidgardSwapHistoryResponse,
+  MidgardTvlHistoryResponse,
+} from 'lib/utils/thorchain/lp/types'
+import { selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const swapHistoryToChartData = (
+  swapHistory: MidgardSwapHistoryResponse | undefined,
+  runePrice: string,
+): SingleValueData<number>[] => {
+  if (!swapHistory) return []
+
+  return swapHistory.intervals.map(interval => {
+    const intervalVolume2FiatUserCurrency = fromThorBaseUnit(interval.totalVolume)
+      .times(runePrice)
+      .toFixed()
+
+    return {
+      time: Number(interval.startTime),
+      value: Number(intervalVolume2FiatUserCurrency),
+    }
+  })
+}
+
+const tvlToChartData = (
+  tvl: MidgardTvlHistoryResponse,
+  runePrice: string,
+  thorchainNotationAssetId: string,
+): SingleValueData<number>[] =>
+  tvl.intervals.map(interval => {
+    const poolDepth = interval.poolsDepth.find(pool => pool.pool === thorchainNotationAssetId)
+    const poolTotalDepth = poolDepth?.totalDepth ?? '0'
+
+    const tvlFiat = fromThorBaseUnit(poolTotalDepth).times(runePrice).toFixed()
+    return {
+      time: Number(interval.startTime),
+      value: Number(tvlFiat),
+    }
+  })
+
+// @ts-ignore we can't make this a partial record as we need to use this as a tuple to spread as useQuery params
+const INTERVAL_PARAMS_BY_INTERVAL: Record<Interval | 'all', [Interval, number]> = {
+  day: ['hour', 24],
+  week: ['hour', 24 * 7],
+  month: ['day', 7 * 4],
+  all: ['month', 24],
+}
+
+type PoolChartProps = {
+  thorchainNotationAssetId: string
+}
+export const PoolChart = ({ thorchainNotationAssetId }: PoolChartProps) => {
+  const [selectedInterval, setSelectedInterval] = useState<Interval | 'all'>('day')
+  const [selectedDataType, setSelectedDataType] = useState<'volume' | 'liquidity'>('volume')
+
+  const setSelectedVolumeDataType = useCallback(() => setSelectedDataType('volume'), [])
+  const setSelectedLiquidityDataType = useCallback(() => setSelectedDataType('liquidity'), [])
+
+  const runeMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, thorchainAssetId),
+  )
+
+  const { data: swapsData } = useQuery({
+    ...reactQueries.midgard.swapsData(
+      thorchainNotationAssetId,
+      ...INTERVAL_PARAMS_BY_INTERVAL[selectedInterval],
+    ),
+    select: data => swapHistoryToChartData(data, runeMarketData.price),
+  })
+
+  const { data: tvl } = useQuery({
```

Comment:
do we want loading state and/or error handling?

Reply:
I didn't add any since the lib doesn't support that natively and we load pretty fast, but if @reallybeard knows how to add some fancy loading state visually, it's all about consuming `isLoading` states from the queries!